// @madzadev/image-slider/dist/index.modern.js
import React, { useState, useEffect } from 'react';
import backButton from './back~QqjScjVe.svg';
import nextButton from './next~FCUOztpN.svg';

var styles = {"wrapper":"_1I_qj","button":"_2hTXI","leftClick":"_mLQYC","rightClick":"_24aXX","dots":"_3MuXE","dot":"_3U2py","activeDot":"_2uiQN"};

var Slider = function Slider(_ref) {
  var imageList = _ref.imageList,
      width = _ref.width,
      height = _ref.height,
      _ref$loop = _ref.loop,
      loop = _ref$loop === void 0 ? true : _ref$loop,
      _ref$autoPlay = _ref.autoPlay,
      autoPlay = _ref$autoPlay === void 0 ? true : _ref$autoPlay,
      _ref$autoPlayInterval = _ref.autoPlayInterval,
      autoPlayInterval = _ref$autoPlayInterval === void 0 ? 3000 : _ref$autoPlayInterval,
      _ref$showArrowControl = _ref.showArrowControls,
      showArrowControls = _ref$showArrowControl === void 0 ? true : _ref$showArrowControl,
      _ref$showDotControls = _ref.showDotControls,
      showDotControls = _ref$showDotControls === void 0 ? true : _ref$showDotControls,
      _ref$bgColor = _ref.bgColor,
      bgColor = _ref$bgColor === void 0 ? "none" : _ref$bgColor;

  var _useState = useState(0),
      active = _useState[0],
      setActive = _useState[1];

  var setPreviousImage = function setPreviousImage() {
    if (active !== 0) {
      setActive(active -= 1);
    } else {
      if (loop) {
        setActive(active = imageList.length - 1);
      }
    }
  };

  var setNextImage = function setNextImage() {
    if (active !== imageList.length - 1) {
      setActive(active += 1);
    } else {
      if (loop) {
        setActive(active = 0);
      }
    }
  };

  var leftClickHandle = function leftClickHandle() {
    setPreviousImage();
  };

  var rightClickHandle = function rightClickHandle() {
    setNextImage();
  };

  var dotClickHandler = function dotClickHandler(e) {
    var dotNum = e.target.getAttribute("data-key");
    setActive(active = parseInt(dotNum));
  };

  useEffect(function () {
    if (autoPlay) {
      var autoSlider = setInterval(setNextImage, autoPlayInterval);
      return function () {
        return clearInterval(autoSlider);
      };
    }
  }, [active]);
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
    className: styles.wrapper,
    style: {
      backgroundColor: bgColor
    }
  }, (showArrowControls && !loop && active !== 0 || showArrowControls && loop) && /*#__PURE__*/React.createElement("div", {
    className: styles.leftClick,
    onClick: leftClickHandle
  }, /*#__PURE__*/React.createElement("img", {
    className: styles.button,
    src: backButton,
    alt: "back"
  })), /*#__PURE__*/React.createElement("img", {
    src: imageList[active].url,
    style: {
      width: width,
      height: height,
      objectFit: "cover"
    },
    alt: "image"
  }), (showArrowControls && !loop && active !== imageList.length - 1 || showArrowControls && loop) && /*#__PURE__*/React.createElement("div", {
    className: styles.rightClick,
    onClick: rightClickHandle
  }, /*#__PURE__*/React.createElement("img", {
    className: styles.button,
    src: nextButton,
    alt: "next"
  }))), showDotControls && /*#__PURE__*/React.createElement("div", {
    className: styles.dots
  }, imageList.map(function (el, index) {
    if (index !== active) {
      return /*#__PURE__*/React.createElement("div", {
        key: index,
        className: styles.dot,
        "data-key": index,
        onClick: dotClickHandler
      });
    } else {
      return /*#__PURE__*/React.createElement("div", {
        key: index,
        className: styles.activeDot
      });
    }
  })));
};

export default Slider;
